tests/test_cice_grid.py

```diff
@@ -0,0 +1,197 @@
+import pytest
+import xarray as xr
+from numpy.testing import assert_allclose
+from numpy import deg2rad
+from subprocess import run
+
+from om3utils.cice_grid import CiceGridNc
+
+# ----------------
+# test data:
+
+
+class MomGrid:
+    """Generate a sample tripole grid to use as test data"""
+
+    def __init__(self, tmp_path):
+        self.path = str(tmp_path) + "/ocean_hgrid.nc"
+        self.mask_path = str(tmp_path) + "/ocean_mask.nc"
+
+        # generate an tripolar grid as test data
+        run(
+            [
+                "ocean_grid_generator.py",
+                "-r",
+                "0.25",  # 4 degree grid
+                "--no_south_cap",
+                "--ensure_nj_even",
+                "-f",
+                self.path,
+            ]
+        )
+
+        # open ocean_hgrid.nc
+        self.ds = xr.open_dataset(self.path)
+
+        # an ocean mask with a arbitrary mask
+        self.mask_ds = xr.Dataset()
+        self.mask_ds["mask"] = (self.ds.area.coarsen(ny=2).sum().coarsen(nx=2).sum()) > 5e9
+        self.mask_ds.to_netcdf(self.mask_path)
+
+
+class CiceGrid:
+    """Make the CICE grid, using script under test"""
+
+    def __init__(self, mom_grid, tmp_path):
+        self.path = str(tmp_path) + "/grid.nc"
+        self.kmt_path = str(tmp_path) + "/kmt.nc"
+        cice_grid = CiceGridNc(self.path, self.kmt_path)
+        cice_grid.build_from_mom(mom_grid.path, mom_grid.mask_path)
+        self.ds = xr.open_dataset(self.path, decode_cf=False)
+        self.kmt_ds = xr.open_dataset(self.kmt_path, decode_cf=False)
+
+
+# pytest doesn't support class fixtures, so we need these two constructor funcs
+@pytest.fixture
+def mom_grid(tmp_path):
+    return MomGrid(tmp_path)
+
+
+@pytest.fixture
+def cice_grid(mom_grid, tmp_path):
+    return CiceGrid(mom_grid, tmp_path)
+
+
+@pytest.fixture
+def test_grid_ds(mom_grid):
+    # this generates the expected answers
+
+    ds = mom_grid.ds
+
+    test_grid = xr.Dataset()
+    # corners of the cice grid are NE corner
+    test_grid["ulat"] = deg2rad(ds.y.isel(nxp=slice(2, None, 2), nyp=slice(2, None, 2)))
+    test_grid["ulon"] = deg2rad(ds.x.isel(nxp=slice(2, None, 2), nyp=slice(2, None, 2)))
```

Comment:
Am I right in guessing that isel slices start from 2 rather than 0 because 0 corresponds to the NE corner of cells that are not part of the actual grid?

Reply:
If so, perhaps add a comment to that effect as it is a bit confusing otherwise

Reply:
> Am I right in guessing that isel slices start from 2 rather than 0 because 0 corresponds to the NE corner of cells that are not part of the actual grid?

Yes that's right, in the old OM2 grid `ulat` starts from zero led to mismatch in longitudes

Reply:
I did a bit of a refactor to hopefully clarify. Its hard to write this stuff concisely in comments.